Lesson5/src/service/MilitaryBase.java

```diff
@@ -0,0 +1,65 @@
+package service;
+
+import model.Person;
+import model.Recruit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MilitaryBase {
+    private int maxSolders;
+    private List<Recruit> recruits = new ArrayList<>();
+
+    public MilitaryBase(int maxSolders) {
+        this.maxSolders = maxSolders;
+    }
+
+    public int getAvailablePlaces() {
+        return maxSolders - recruits.size();
+    }
+
+    private void checkForTheMatchingRecruits(List<Person> fitForArmy) {
+        for (Person person : fitForArmy) {
+            for (Person recruit : recruits) {
+                if (person == recruit ||
+                        person.getName().equals(recruit.getName()) &&
+                                person.getAge().equals(recruit.getAge()) &&
+                                person.getHeight().equals(recruit.getHeight())) {
+                    System.out.println(person.getName() + " are serving in the army now");
+                } else {
+                    addSolder(person);
```

Comment:
this will not work. You can't modify collection that you are iterating through right now.

You have   for (Person recruit : recruits) and then in addSolder you do recruits.add(new Recruit. It will cause an error.

First you need to go through all recruits and verify that NONE of them is equal to person, and then if it's true -- you can add person to the recruits